### PR TITLE
UndertowHttp2Client and test cases

### PR DESCRIPTION
**Pull request by stevehu**
There are still some debug info shows up but all test cases are executed successfully. Let me know if anything needs to be updated. Thanks.

### CODE REVIEW

**Comment by undertow-pull-request** [comment]
Can one of the admins verify this patch?

File: core/src/main/java/io/undertow/client/UndertowHttp2Client.java [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+import java.util.Map;
+import java.util.ServiceLoader;
+
+/**
+ * Undertow http/2 client class. This class loads {@link ClientProvider} implementations, and uses them to
+ * create connections to a target.
+ *
+ * @author Steve Hu
+ */
+public class UndertowHttp2Client {
```
Comment (stuartwdouglas) [inline core/src/main/java/io/undertow/client/UndertowHttp2Client.java]: What is the purpose of this class? The standard HTTP client can handle HTTP/2 connections?

**Comment by stevehu** [comment]
I debugged into the UndertowClient and found that it can only load HTTP/1.1 ClientProviders with the key of "http" or "https". HTTP/2 providers have key as "h2", "h2c" and "h2-prior" and they cannot be loaded based from the following line.

```
ClientProvider provider = clientProviders.get(uri.getScheme());
```

**Comment by stuartwdouglas** [comment]
If you want to force h2 then you need to use URI's such as 'h2://localhost:8443', however if you pass the ENABLE_HTTP2 option into the client then the HTTP/1.1 provider will attempt to upgrade to H2 (see the impl at https://github.com/undertow-io/undertow/blob/master/core/src/main/java/io/undertow/client/http/HttpClientConnection.java#L139).

**Comment by stevehu** [comment]
Great!!!! Thanks Stuart for the clarification. I missed the upgrade logic when reading the code.

**Comment by stuartwdouglas** [comment]
I am still happy to include the test, I just don't really think we need a separate HTTP/2 client.

**Comment by stevehu** [comment]
Cool. I will added some test cases to the existing one and give you another pull request. Thanks.

**Pull request closed without merging**